tests/unit/Alignment.cpp

```diff
@@ -43,6 +43,18 @@ void testAlignment(const model::QualifiedType &Type,
   }
 }
 
+template<typename... Types>
+  requires(std::convertible_to<Types, uint64_t> && ...)
+bool ABIhasNeededScalars(const abi::Definition &ABI, Types... Values) {
+  return (llvm::is_contained(ABI.ScalarTypes(), Values) && ...);
+}
+
+template<typename... Types>
+  requires(std::convertible_to<Types, uint64_t> && ...)
+bool ABIhasNeededFloats(const abi::Definition &ABI, Types... Values) {
+  return (llvm::is_contained(ABI.FloatingPointScalarTypes(), Values) && ...);
+}
+
```

Comment:
Please describe what's going on in this commit, I'm confused.

Reply:
If you remember, scalar alignment requirements are now a lot stricter. Which lead this test to fail.

This is pretty much some extra machinery to be able to express "if ABI has no such prototype, skip a test".

Reply:
1. Use `SortedVector::contains` instead of `llvm::is_contained`.
2. Avoid templates, too big
3. Wouldn't a more appropriate name be `hasFloatOfSize(...)`? `ABIhasNeededFloats` does not resonate a lot with me.

Reply:
1. 👌
2. It's the usual dilemma of "do you want callee to be ugly? Or the caller?" and I prefer the callee. This let's caller just say
```cpp
if (ABIhasFloatsOfSizes(Size1, Size2, Size3))
  do_something(Size1, Size2, Size3);
```

instead of doing ugly stuff like
```cpp
if (ABIhasFloatOfSize(Size1))
  if (ABIhasFloatOfSize(Size2))
    if (ABIhasFloatOfSize(Size3))
      do_something(Size1, Size2, Size3);
```

3. `ABIhasFloatsOfSizes` works (without plurality if we drop the multi-check version :sadparrot:).

Reply:
2. I'd just take an `ArrayRef<uint64_t>` and use `std::any_of` or something similar

Reply:
```cpp
if (ABIhasFloatsOfSizes(ABI, { Size1, Size2, Size3 }))
  do_something(ABI, Size1, Size2, Size3);
```

is still pretty ugly, but I guess I can live with it.

Reply:
Done (I went with the `std::initializer_list`)

Also, while at it, I renamed `ABIhasScalarsOfSizes` into `ABIhasIntsOfSizes` because, technically `Floats` are also `Scalars`.